concent_api/core/validation.py

```diff
@@ -218,3 +226,121 @@ def validate_list_task_to_compute_ids(subtask_results_accepted_list):
     for task_to_compute in subtask_results_accepted_list:
         subtask_ids.append(task_to_compute.subtask_id + ':' + task_to_compute.task_id)
     return len(subtask_ids) == len(set(subtask_ids))
+
+
+def get_validated_client_public_key_from_client_message(golem_message: message.base.Message):
+    if (
+        isinstance(golem_message, message.concents.ForcePayment) and
+        isinstance(golem_message.subtask_results_accepted_list, list) and
+        len(golem_message.subtask_results_accepted_list) > 0
+    ):
+        task_to_compute = get_field_from_message(golem_message.subtask_results_accepted_list[0], 'task_to_compute')
+    elif isinstance(golem_message, message.base.Message):
+        task_to_compute = get_field_from_message(golem_message, 'task_to_compute')
+    else:
+        return None
+
+    if task_to_compute is not None:
+        if isinstance(golem_message, (
+            message.ForceReportComputedTask,
+            message.concents.ForceSubtaskResults,
+            message.concents.ForcePayment,
+            message.concents.SubtaskResultsVerify,
+        )):
+            client_public_key = task_to_compute.provider_public_key
+            validate_public_key(client_public_key, 'provider_public_key')
+        elif isinstance(golem_message, (
+            message.AckReportComputedTask,
+            message.RejectReportComputedTask,
+            message.concents.ForceGetTaskResult,
+            message.concents.ForceSubtaskResultsResponse,
+        )):
+            client_public_key = task_to_compute.requestor_public_key
+            validate_public_key(client_public_key, 'requestor_public_key')
+
+        return client_public_key
+
+    return None
+
+
+def validate_file_transfer_token(file_transfer_token: message.concents.FileTransferToken):
```

Comment:
I dislike this function. As it is validation function it should not return anything and rasie exception in case it detects there's something wrong with `FileTransferToken`

Reply:
Done